src/textual/widgets/_input.py

```diff
@@ -421,10 +421,11 @@ async def _on_click(self, event: events.Click) -> None:
         cell_offset = 0
         _cell_size = get_character_cell_size
         for index, char in enumerate(self.value):
-            if cell_offset >= click_x:
+            cell_width = _cell_size(char)
+            if cell_offset <= click_x < (cell_offset + cell_width):
```

Comment:
NOTE TO REVIEWER: This line is equivalent to:
```suggestion
            if (click_x - cell_offset) < cell_width:
```
This option is shorter and does just one comparison instead of two.
However, I eventually preferred the former because it easier to understand in my opinion.

If any of the maintainers prefer the latter, please let me know.